update-vo-config/update-vo-config

```diff
@@ -0,0 +1,507 @@
+
+
+def to_base26(integer):
+    """Base26 digits are normally from 0 to P but the returned string
+    is containing only letters:
+    digits from 0 to 9 are replaced by letters from q to z.
+    The returned string is in lowercase
+    """
+    div = integer & 0xFFFFFFFF
+    base26 = ""
+    while True:
+        (div, mod) = divmod(div, 26)
+        # Replace digits from 0 to 9 by letters from Q to Z
+        if mod < 10:
+            mod = mod + ord('P') + 1
+        else:
+            mod = mod + ord('A') - 10
+        base26 = chr(mod) + base26
+        if div < 1:
+            break
+    return base26.lower()
+
+
+def get_legacy_suffix(vo_id, fqan, account_suffix, debug=False):
+    """Generated suffix is based on base26-like conversion of FQAN length
+       and VO ID. Despite this is a very bad choice for uniqueness, it is
+       impossible to change without breaking backward compatibility of
+       generated accounts. New algorithm is implemented as a distinct method.
+    """
+    if fqan['type'] in FQAN_TYPE_MAPPING:
+        return FQAN_TYPE_MAPPING[fqan['type']]
+    else:
+        is_unique = False
+        legacy_suffix = ""
+        j = 0
+        while not is_unique:
+            if debug and j:
+                sys.stdout.write("Debug: Suffix '" + legacy_suffix +
+                                 "' not unique for FQAN " + fqan['expr'] +
+                                 " (attempt " + j + "\n")
+            legacy_suffix = to_base26(len(fqan['expr']) + j * 100)
+            legacy_suffix += to_base26(vo_id)
+            if legacy_suffix not in account_suffix:
+                is_unique = True
+            j += 1
+    return legacy_suffix
+
+
+def get_account_suffix(vo_name, fqan):
+    """Returns either a predefined suffix for pilot, production and software
+       manager roles, or a 3-character suffix corresponding to the base26-like
+       encoding of the FQAN hashcode
+    """
+    if fqan['type'] in FQAN_TYPE_MAPPING:
+        return FQAN_TYPE_MAPPING[fqan['type']]
+    else:
+        relative_fqan = fqan['expr'].replace('/' + vo_name, '', 1)
+        return to_base26(java_string_hashcode(relative_fqan))
+
+
+def write_cert_info(voms_endpoint):
+    """Writes subject and issuer of VOMS server valid certificates as a
+       dict element
+    """
+    content = ""
+    subject = ""
+    issuer = ""
+    if voms_endpoint.cert:
+        cert = crypto.load_certificate(crypto.FILETYPE_PEM, voms_endpoint.cert)
+        for (field, value) in cert.get_subject().get_components():
+            subject = subject + "/" + field + "=" + value
+        for (field, value) in cert.get_issuer().get_components():
+            issuer = issuer + "/" + field + "=" + value
+        content = format("    '%s', dict(\n" % voms_endpoint.hostname)
+        content += format("        'subject', '%s',\n" % subject)
+        content += format("        'issuer', '%s',\n" % issuer)
+        content += format("    ),\n")
```

Comment:
It feels like this format string should be defined as a multi-line constant at the top of the file, e.g.

```python
VOMS_ENDPOINT_CERT_TEMPLATE = '''
    '%s', dict(
        'subject', '%s',
        'issuer', '%s',
   ),
'''
```